end_to_end/end_to_end_suite_test.go

```diff
@@ -2195,6 +2195,9 @@ LANGUAGE plpgsql NO SQL;`)
 					numSegments := dbconn.MustSelectString(restoreConn, "SELECT numsegments FROM gp_distribution_policy where localoid = 'schemaone.test_table'::regclass::oid")
 					Expect(numSegments).To(Equal(strconv.Itoa(segmentCount)))
 
+					// check there is no pipe errors on segments
+					errSegments := dbconn.MustSelectString(restoreConn, fmt.Sprintf("SELECT exists (SELECT * FROM gp_toolkit.__gp_log_segment_ext WHERE logdatabase = current_database() AND logmessage LIKE 'read err msg from pipe%%_%06d_%%')", gprestoreCmd.Process.Pid))
```

Comment:
Using `SELECT *` is bad idea.

Reply:
```
SELECT true AS exists FROM gp_toolkit.__gp_log_segment_ext WHERE logdatabase = current_database() AND logmessage LIKE 'read err msg from pipe%%_%06d_%%' limit 1;
```

Reply:
Why? This query contains exists, not only select *!
```sql
explain verbose select exists (select * from gp_toolkit.__gp_log_segment_ext where logmessage like 'read err msg from pipe%');
                                                             QUERY PLAN                                                              
-------------------------------------------------------------------------------------------------------------------------------------
 Result  (cost=0.00..943633.88 rows=1 width=1)
   Output: (COALESCE((count((count()))), '0'::bigint) > '0'::bigint)
   ->  Result  (cost=0.00..943633.88 rows=1 width=8)
         Output: (count((count())))
         ->  Nested Loop Left Join  (cost=0.00..943633.88 rows=1 width=8)
               Output: (count((count())))
               Join Filter: true
               ->  Result  (cost=0.00..0.00 rows=1 width=1)
                     Output: true
               ->  Materialize  (cost=0.00..490.52 rows=1 width=8)
                     Output: (count((count())))
                     ->  Aggregate  (cost=0.00..490.52 rows=1 width=8)
                           Output: count((count()))
                           ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..490.52 rows=1 width=8)
                                 Output: (count())
                                 ->  Aggregate  (cost=0.00..490.52 rows=1 width=8)
                                       Output: count()
                                       ->  External Scan on gp_toolkit.__gp_log_segment_ext  (cost=0.00..490.52 rows=133334 width=1)
                                             Filter: (__gp_log_segment_ext.logmessage ~~ 'read err msg from pipe%'::text)
 Optimizer: Pivotal Optimizer (GPORCA)
(20 rows)

set optimizer = off;
SET
explain verbose select exists (select * from gp_toolkit.__gp_log_segment_ext where logmessage like 'read err msg from pipe%');
                                                QUERY PLAN                                                
----------------------------------------------------------------------------------------------------------
 Result  (cost=0.27..0.28 rows=1 width=0)
   Output: $0
   InitPlan 1 (returns $0)  (slice2)
     ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..13500.00 rows=50000 width=0)
           ->  External Scan on gp_toolkit.__gp_log_segment_ext  (cost=0.00..13500.00 rows=16667 width=0)
                 Filter: (__gp_log_segment_ext.logmessage ~~ 'read err msg from pipe%'::text)
 Optimizer: Postgres query optimizer
 Settings: optimizer=off
(8 rows)
```

Reply:
If you remove `*`, query will return the same results/

Reply:
Query optimization in tests is not critical for the task with blocker priority